ios/FluentUI/People Picker/AvatarStack.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+@objc(MSFAvatarStack)
+open class AvatarStack: UIView {
+    /// Array of avatars to display in the avatar stack.
+    @objc open var avatars: [Avatar] {
+        didSet {
+            updateAvatars()
+        }
+    }
+
+    /// The size for avatar views in the avatar stack
+    @objc open var avatarSize: AvatarSize {
+        didSet {
+            updateAvatars()
+        }
+    }
+
+    /// Set to true to display avatar borders in the avatar stack.
+    @objc open var displaysBorders: Bool {
+        didSet {
+            updateAvatars()
+        }
+    }
+
+    @objc public required init(with avatars: [Avatar], size: AvatarSize, displaysBorders: Bool = false) {
+        self.avatars = avatars
+        self.avatarSize = size
+        self.displaysBorders = displaysBorders
+        super.init(frame: .zero)
+
+        updateAvatars()
+    }
+
+    @available(*, unavailable)
+    @objc public required override init(frame: CGRect) {
+        preconditionFailure("init(frame:) has not been implemented")
+    }
+
+    @available(*, unavailable)
+    @objc public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    open override var intrinsicContentSize: CGSize {
+        let avatarsCount = CGFloat(avatars.count)
+        var width = avatarSize.size.width * avatarsCount
+        let avatarOverlap = self.avatarOverlap()
+
+        if avatarsCount > 1 {
+            width -= (avatarsCount - 1) * avatarOverlap
+        }
+
+        return CGSize(width: width, height: avatarSize.size.height)
+    }
+
+    open override func layoutSubviews() {
+        super.layoutSubviews()
+        updateLayerMask()
+    }
+
+    private struct Constants {
+        static let avatarOverlapRatio: CGFloat = 0.14
+    }
+
+    private var avatarViews: [AvatarView] = []
+
+    private func updateAvatars() {
+        removeAllSubviews()
+
+        avatarViews.removeAll()
+        var constraints: [NSLayoutConstraint] = []
+        var previousAvatarView: AvatarView?
+        let avatarOverlap = self.avatarOverlap()
+
+        for avatar in avatars {
```

Comment:
does this logic work with RTL? shouldn't the stack go other way around?

Reply:
It mostly does, I did find a small issue. I'll send an update.